File: src/main/java/com/app/umkaSchool/model/DailyChallenge.java
package com.app.umkaSchool.model;

import com.fasterxml.jackson.annotation.JsonIgnore;
import jakarta.persistence.*;
import lombok.Data;
import lombok.EqualsAndHashCode;
import lombok.ToString;
import org.hibernate.annotations.CreationTimestamp;
import org.hibernate.annotations.UpdateTimestamp;

import java.time.LocalDate;
import java.time.ZonedDateTime;
import java.util.HashSet;
import java.util.Set;
import java.util.UUID;

@Data
@Entity
@Table(name = "daily_challenge", schema = "school")
public class DailyChallenge {
    @Id
    @GeneratedValue(strategy = GenerationType.UUID)
    @Column(name = "daily_challenge_id", nullable = false)
    private UUID id;

    @Column(name = "challenge_date", nullable = false, unique = true)
    private LocalDate challengeDate;

    @Column(nullable = false)
    private String title;

    private String description;

    @JsonIgnore
    @ToString.Exclude
    @EqualsAndHashCode.Exclude
    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "created_by")
    private Teacher createdBy;

    @CreationTimestamp
    @Column(name = "created_at", nullable = false, updatable = false)
    private ZonedDateTime createdAt;

    @UpdateTimestamp
    @Column(name = "updated_at", nullable = false)
    private ZonedDateTime updatedAt;

    @JsonIgnore
    @ToString.Exclude
    @EqualsAndHashCode.Exclude
    @OneToMany(mappedBy = "dailyChallenge", cascade = CascadeType.ALL)
    private Set<DailyChallengeExercise> exercises = new HashSet<>();
}

File: frontend/src/types/dailyChallenge.ts
export interface DailyChallenge {
  id: string;
  challengeDate: string;
  title: string;
  description?: string;
  createdById?: string;
  createdByName?: string;
  createdAt: string;
  updatedAt: string;
  exercises: DailyChallengeExercise[];
}

export interface DailyChallengeExercise {
  exerciseId: string;
  exerciseTypeId: string;
  exerciseTypeName: string;
  parameters: string;
  difficulty: number;
  points: number;
  orderIndex: number;
}

export interface CreateDailyChallengeRequest {
  challengeDate: string;
  title: string;
  description?: string;
  createdById?: string;
  exercises: ExerciseRequest[];
}

export interface ExerciseRequest {
  exerciseId: string;
  orderIndex?: number;
}

